### DISCUSSION TITLE
Disabled HTTP compression but some responses are still being compressed

### DISCUSSION THREAD

**Discussion by tuhlaajapoika** [seerr-team/seerr#1113] [Q&A] [answered]
Hello,

My Jellyseerr instance is running behind an NGINX reverse proxy that handles HTTP data compression. I've patched and built Jellyseerr to prevent it from compressing before passing them to the reverse proxy. I modified `next.config.js` as described on the [Next.js documentation](https://nextjs.org/docs/app/api-reference/next-config-js/compress#disabling-compression), which worked fine before v2.x.

However, after v2.x, some responses are still being precompressed and I can't figure out which part of the code (or its dependencies) is applying the compression. Data that is being compressed is provided by Express.

I'm unsure how to identify the changes made on v2.x onwards to adjust my patches accordingly. I'm relatively new to Node.js applications and the list of dependencies is a bit overwhelming to be honest. Should I make changes to `server/index.ts` to explicitly disable compression?

Patch for `next.config.js`

```
diff --unified --recursive --text --color a/next.config.js b/next.config.js
--- a/next.config.js	2024-06-13 14:28:59.908618620 +0300
+++ b/next.config.js	2024-06-13 14:29:52.302294307 +0300
@@ -21,6 +21,7 @@
 
     return config;
   },
+  compress: false,
   experimental: {
     scrollRestoration: true,
     largePageDataBytes: 256000,
```

**Comment by gauthier-th** [comment]
Did you delete the dist, .next, and node_modules directories before rebuilding Jellyseerr?
What responses are still compressed?

**Comment by tuhlaajapoika** [comment]
Sorry for the late response.

It was a fresh build on a cloud server. After testing throughly with curl it seems that Jellyseerr works as it should. It was a misunderstanding from my part. Firefox's developer tools keeps showing "Content-Encoding: gzip" on random .css and .js files so it might be a response from a CDN or a another server. Curl does not receive Content-Encoding header for the same URL.

Thank you for the reply, though.

**Comment by tuhlaajapoika** [comment]
@gauthier-th I decided to get back to this. Sorry for tagging.

I'm trying to offload compression to NGINX and use caching (I'm not sure if it affects performance). Every response for `/_next/static` gets compressed. Seems like something else does the compression and not next.js, because `compress: false` does not affect it.

I've search every file with grep for compression related lines and I'm out of ideas. Responses for `/_next/static` are the slowest which contains css and js files mostly.

**Comment by gauthier-th** [comment]
> Seems like something else does the compression and not next.js

I can't think of anything that would compress this other than Next.js.
We use Express.js for the Jellyseerr API and not Next.js, but gzip compression for Express.js is not enabled by default.

Are you sure you recompiled the Next.js app correctly? Deleted the `.next`, `dist` and `node_modules` folders, run `pnpm i` and then `pnpm build`?

**Reply by tuhlaajapoika** [reply, marked as answer]
I changed my script to build using the git repo instead of the source tar from releases. And guess what I realised while editing the script, I had forgotten to copy the patched `next.config.js` to the installation directory... All good now, nothing gets compressed.

Thank you for you time, though.
